examples/seq2seq/distillation.py

```diff
@@ -156,36 +185,38 @@ def _step(self, batch):
         def zero_tensor():
             return torch.tensor(0.0).type_as(student_lm_loss)
 
+        teacher_enc_outputs = enc_outputs
         hid_loss_enc, hid_loss_dec = zero_tensor(), zero_tensor()
         if self.different_encoder:  # compute encoder hidden state loss
             with torch.no_grad():
-                teacher_enc_hid = self.teacher.get_encoder()(
-                    input_ids, attention_mask=src_mask, output_hidden_states=True, return_dict=True
-                ).hidden_states
-
-            hid_loss_enc = self.calc_hidden_loss(
+                teacher_enc_outputs, teacher_enc_hid = self.teacher.get_encoder()(
+                    input_ids, attention_mask=src_mask, output_hidden_states=True
+                )
+            hid_loss_enc = self.maybe_calc_hidden_loss(
```

Comment:
It would be great to move the decision about whether to calculate hidden loss to `__init__`.
I think that `self.different_encoder` should be `False` in your case, so we shouldn't need an `if` statement in `calc_hidden_loss`.

Reply:
Hi Sam, thanks for your review! Actually `self.different_encoder` is generally `True` when the teacher and student are initialized with different base models, for e.g., when student base model is `t5_small` and teacher base model is `t5_large`, `student_encoder_layers` is 6 and `teacher_encoder_layers` is 24. Because they have different architectures they have different number of layers.

Reply:
OK, let's make one flag that is set in `__init__` that is roughly
```python
self.do_hidden_loss  =self.different_encoder or self.student is None
```
and just check that.

Reply:
done